### ISSUE TITLE
[Math] The `math_gtest` is failing on on Mac OS build

### ISSUE THREAD

**Issue by pbukva** [fetchai/ledger#1192]
The _math_gtest_ is failing on on Mac OS (at least in Debug build):
`15/33 Test #17: math_gtest ........................***Exception: Bus error  0.01 sec`

The issue does not manifest on Linux build.

Bellow is the callstack when the failure happens (copy-paste from lldb):

```sh
fetch::fixed_point::FixedPoint<(unsigned short)32, (unsigned short)32>::FixedPoint(fetch::fixed_point::FixedPoint<(unsigned short)32, (unsigned short)32> const&) fixed_point.hpp:739
fetch::fixed_point::FixedPoint<(unsigned short)32, (unsigned short)32>::FixedPoint(fetch::fixed_point::FixedPoint<(unsigned short)32, (unsigned short)32> const&) fixed_point.hpp:740
fetch::fixed_point::FixedPoint<(unsigned short)32, (unsigned short)32>::operator|(fetch::fixed_point::FixedPoint<(unsigned short)32, (unsigned short)32> const&) const fixed_point.hpp:1247
::__cxx_global_var_init.50() fixed_point.hpp:664
ImageLoaderMachO::doModInitFunctions(ImageLoader::LinkContext const&) 0x000000010d741592
ImageLoaderMachO::doInitialization(ImageLoader::LinkContext const&) 0x000000010d741798
ImageLoader::recursiveInitialization(ImageLoader::LinkContext const&, unsigned int, char const*, ImageLoader::InitializerTimingList&, ImageLoader::UninitedUpwards&) 0x000000010d73cbea
ImageLoader::processInitializers(ImageLoader::LinkContext const&, unsigned int, ImageLoader::InitializerTimingList&, ImageLoader::UninitedUpwards&) 0x000000010d73bd73
ImageLoader::runInitializers(ImageLoader::LinkContext const&, ImageLoader::InitializerTimingList&) 0x000000010d73be05
dyld::initializeMainExecutable() 0x000000010d72b765
dyld::_main(macho_header const*, unsigned long, int, char const**, char const**, char const**, unsigned long*) 0x000000010d730709
dyldbootstrap::start(macho_header const*, int, char const**, long, macho_header const*, unsigned long*) 0x000000010d72a503
_dyld_start 0x000000010d72a036
```

**Comment by wkaluza** [comment]
I tracked down the problem to commit 06229c4c24d0674850441f03df6fd75952ca9881: Feature/fixedpoint trig functions (#1098). Removing it from master (currently at 41729d71741f56a4d6286076164351425f8a8d55) using interactive rebase results in a clean Mac debug build with no issues.

@markos do you have any suspicions what may be the cause?

**Comment by markos** [comment]
I think this should be fixed with #1244
